pkg/kubelet/dockershim/exec.go

```diff
@@ -114,22 +112,24 @@ func (*NativeExecHandler) ExecInContainer(client libdocker.Interface, container
 		return err
 	}
 
-	// if ExecProbeTimeout feature gate is disabled, preserve existing behavior to ignore exec timeouts
-	var execTimeout <-chan time.Time
+	execCtx := ctx
```

Comment:
Problem is that StartExec is actually a blocking call, so turns out the for loop below calling `InspectExec` was never actually called on a running process.